src/managedwriter/encoder.ts

```diff
@@ -87,47 +87,73 @@ export class JSONEncoder {
     return this._type.encode(msg).finish();
   }
 
-  private convertRow(source: any): Object {
+  private convertRow(source: any, ptype: protobuf.Type): Object {
     const row = extend(true, {}, source);
     for (const key in row) {
       const value = row[key];
       if (value === null) {
         continue;
       }
-      if (value instanceof Date) {
-        const pfield = this._type.fields[key];
-        if (!pfield) {
-          continue;
-        }
-        switch (pfield.type) {
-          case 'int32': // DATE
-            // The value is the number of days since the Unix epoch (1970-01-01)
-            row[key] = value.getTime() / (1000 * 60 * 60 * 24);
-            break;
-          case 'int64': // TIMESTAMP
-            // The value is given in microseconds since the Unix epoch (1970-01-01)
-            row[key] = value.getTime() * 1000;
-            break;
-          case 'string': // DATETIME
-            row[key] = value.toJSON().replace(/^(.*)T(.*)Z$/, '$1 $2');
-            break;
-        }
+      const encodedValue = this.encodeRowValue(value, key, ptype);
+      if (encodedValue === undefined) {
         continue;
       }
-      if (Array.isArray(value)) {
-        row[key] = value.map(v => {
-          if (!this.isPlainObject(v)) {
-            return v;
-          }
-          return this.convertRow(v);
-        });
-        continue;
+      row[key] = encodedValue;
+    }
+    return row;
+  }
+
+  private encodeRowValue(
+    value: any,
+    key: string,
+    ptype: protobuf.Type
+  ): any | null {
```

Comment:
you are returning `undefined` but this line says `any | null`.  it works because `any` allows anything, but maybe it's better to say `any | undefined`?

Reply:
ah good catch, I started using `null` and later moved to `undefined` and forgot to change the signature.